## `nixpkgs-review` result

Generated using [`nixpkgs-review`](https://github.com/Mic92/nixpkgs-review).

Command: `nixpkgs-review pr 445425`
Commit: `af2ef72797a67ac9324fd60926c37a3a81a7ed66`

---
### `x86_64-linux`
<details>
  <summary>:x: 4 packages failed to build:</summary>
  <ul>
    <li>python312Packages.pytradfri</li>
    <li>python312Packages.pytradfri.dist</li>
    <li>python313Packages.pytradfri</li>
    <li>python313Packages.pytradfri.dist</li>
  </ul>
</details>
<details>
  <summary>:white_check_mark: 17 packages built:</summary>
  <ul>
    <li>expliot</li>
    <li>expliot.dist</li>
    <li>home-assistant-component-tests.homekit_controller</li>
    <li>home-assistant-component-tests.tradfri</li>
    <li>home-assistant-custom-components.philips_airpurifier_coap</li>
    <li>python312Packages.aioairctrl</li>
    <li>python312Packages.aioairctrl.dist</li>
    <li>python312Packages.aiocoap</li>
    <li>python312Packages.aiocoap.dist</li>
    <li>python312Packages.aiohomekit</li>
    <li>python312Packages.aiohomekit.dist</li>
    <li>python313Packages.aioairctrl</li>
    <li>python313Packages.aioairctrl.dist</li>
    <li>python313Packages.aiocoap</li>
    <li>python313Packages.aiocoap.dist</li>
    <li>python313Packages.aiohomekit</li>
    <li>python313Packages.aiohomekit.dist</li>
  </ul>
</details>

---
<details>
<summary>Error logs: `x86_64-linux`</summary>
<details>
<summary>python312Packages.pytradfri</summary>
<pre>
tests/test_gateway.py::test_gateway_info
  /build/source/tests/test_gateway.py:63: DeprecationWarning: datetime.datetime.utcfromtimestamp() is deprecated and scheduled for removal in a future version. Use timezone-aware objects to represent datetimes in UTC: datetime.datetime.fromtimestamp(timestamp, datetime.UTC).
    assert gateway_info.current_time == datetime.utcfromtimestamp(1509788799)

tests/test_gateway.py::test_gateway_info
  /build/source/pytradfri/gateway.py:317: DeprecationWarning: datetime.datetime.utcfromtimestamp() is deprecated and scheduled for removal in a future version. Use timezone-aware objects to represent datetimes in UTC: datetime.datetime.fromtimestamp(timestamp, datetime.UTC).
    return datetime.utcfromtimestamp(self.raw.first_setup)

tests/test_gateway.py::test_gateway_info
  /build/source/tests/test_gateway.py:65: DeprecationWarning: datetime.datetime.utcfromtimestamp() is deprecated and scheduled for removal in a future version. Use timezone-aware objects to represent datetimes in UTC: datetime.datetime.fromtimestamp(timestamp, datetime.UTC).
    assert gateway_info.first_setup == datetime.utcfromtimestamp(1509474847)

-- Docs: https://docs.pytest.org/en/stable/how-to/capture-warnings.html
=========================== short test summary info ============================
FAILED tests/api/test_aiocoap_api.py::test_request_returns_single - Failed: async def functions are not natively supported.
FAILED tests/api/test_aiocoap_api.py::test_request_returns_list - Failed: async def functions are not natively supported.
FAILED tests/api/test_aiocoap_api.py::test_context_shutdown_request - Failed: async def functions are not natively supported.
FAILED tests/api/test_aiocoap_api.py::test_shutdown_after_reset_protocol_error - Failed: async def functions are not natively supported.
================== 4 failed, 168 passed, 8 warnings in 0.23s ===================</pre>
</details>
<details>
<summary>python313Packages.pytradfri</summary>
<pre>
tests/test_gateway.py::test_gateway_info
  /build/source/tests/test_gateway.py:63: DeprecationWarning: datetime.datetime.utcfromtimestamp() is deprecated and scheduled for removal in a future version. Use timezone-aware objects to represent datetimes in UTC: datetime.datetime.fromtimestamp(timestamp, datetime.UTC).
    assert gateway_info.current_time == datetime.utcfromtimestamp(1509788799)

tests/test_gateway.py::test_gateway_info
  /build/source/pytradfri/gateway.py:317: DeprecationWarning: datetime.datetime.utcfromtimestamp() is deprecated and scheduled for removal in a future version. Use timezone-aware objects to represent datetimes in UTC: datetime.datetime.fromtimestamp(timestamp, datetime.UTC).
    return datetime.utcfromtimestamp(self.raw.first_setup)

tests/test_gateway.py::test_gateway_info
  /build/source/tests/test_gateway.py:65: DeprecationWarning: datetime.datetime.utcfromtimestamp() is deprecated and scheduled for removal in a future version. Use timezone-aware objects to represent datetimes in UTC: datetime.datetime.fromtimestamp(timestamp, datetime.UTC).
    assert gateway_info.first_setup == datetime.utcfromtimestamp(1509474847)

-- Docs: https://docs.pytest.org/en/stable/how-to/capture-warnings.html
=========================== short test summary info ============================
FAILED tests/api/test_aiocoap_api.py::test_request_returns_single - Failed: async def functions are not natively supported.
FAILED tests/api/test_aiocoap_api.py::test_request_returns_list - Failed: async def functions are not natively supported.
FAILED tests/api/test_aiocoap_api.py::test_context_shutdown_request - Failed: async def functions are not natively supported.
FAILED tests/api/test_aiocoap_api.py::test_shutdown_after_reset_protocol_error - Failed: async def functions are not natively supported.
================== 4 failed, 168 passed, 8 warnings in 0.22s ===================</pre>
</details>
</details>

--- a/pkgs/development/python-modules/pytradfri/default.nix
+++ b/pkgs/development/python-modules/pytradfri/default.nix
@@ -6,6 +6,7 @@
   aiocoap,
   dtlssocket,
   pydantic,
+  pytest-asyncio,
   pytestCheckHook,
 }:
 
@@ -32,7 +33,11 @@ buildPythonPackage rec {
     ];
   };
 
-  nativeCheckInputs = [ pytestCheckHook ] ++ optional-dependencies.async;
+  nativeCheckInputs = [
+    pytest-asyncio
+    pytestCheckHook
+  ]
+  ++ optional-dependencies.async;
 
   pythonImportsCheck = [ "pytradfri" ];
 

